### ISSUE TITLE
make citar-org support embark-dwim

### ISSUE THREAD

**Issue by tshu-w** [bdarcus/citar#387]
currently if I run `embark-dwim` on org citation, I got
```
  nil("peeters_cross-language_2021")
  #f(compiled-function () #<bytecode 0x1a7da90d012d516>)()
  embark--quit-and-run(#f(compiled-function () #<bytecode 0x1a7da90d012d516>))
  embark--act(nil (:orig-type oc-citation :orig-target "peeters_cross-language_2021" :bounds nil :type oc-citation :target "peeters_cross-language_2021") t)
  embark-dwim(nil)
  #<subr funcall-interactively>(embark-dwim nil)
  apply(#<subr funcall-interactively> (embark-dwim nil))
  funcall-interactively(embark-dwim nil)
  #<subr call-interactively>(embark-dwim nil nil)
  apply(#<subr call-interactively> (embark-dwim nil nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> embark-dwim nil nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (embark-dwim nil nil))
  call-interactively(embark-dwim nil nil)
  command-execute(embark-dwim)
```
should we add default action(ENTER) in citar-org-map?

**Comment by tshu-w** [comment]
> > currently if I run embark-dwim on org citation, I got
> 
> Seems there's other things wrong with your config then ;-)
> 
> It works for me, and mouse-clicking on a citation runs it, while right-clicking runs `embark-act`.
> 
> > should we add default action(ENTER) in citar-org-map?
> 
> Probably.
> 
> IIRC, I removed it awhile back before there was some bug with it I couldn't figure out.

`embark-act` work, not `embark-dwim`

**Comment by bdarcus** [comment]
Oh, you're right: i can reproduce that!

So `embark-act` and `citar-dwim` work correctly; it's just not `embark-dwim`.

Should probably be an easy fix.

**Comment by bdarcus** [comment]
> Should probably be an easy fix.

I played with this a bit, and couldn't get it work.

Will try again later.

**Comment by tshu-w** [comment]
I originally thought that adding the ENTER binding would be enough, but it seems more complicated than I thought 😂

**Comment by tshu-w** [comment]
> I originally thought that adding the ENTER binding would be enough, but it seems more complicated than I thought 😂

Yes, `mbark--default-action` look `RET` in the action keymap
https://github.com/oantolin/embark/blob/master/embark.el#L1754

**Comment by bdarcus** [comment]
I know, but it doesn't work here.

**Comment by tshu-w** [comment]
i will look at it if i have time

**Comment by bdarcus** [comment]
While you're working on #394, @roshanshariff, can you see if you can figure this out?

**Comment by roshanshariff** [comment]
This should be fixed by #394. Currently `citar-org-map` doesn't have a binding for `RET`, so there's no default action. In #394 we remove the Org-specific keymap in favour of using `citar-buffer-map` (which is renamed to `citar-citation-map`). That has a default action `citar-run-default-action`.
